### PR TITLE
Add python_webapp dockerfile

### PR DESCRIPTION
**Pull request by tschuy**
Allows us to create, say, a GWM docker file like so:

```
FROM osuosl/python_webapp

# GWM specific stuff
RUN sed  -i 's/WSGIPATH/ganeti_webmgr.ganeti_web/g' /etc/supervisor.d/app.conf
COPY ./ /opt/app/src
COPY ./config.yml /opt/ganeti_webmgr/config/config.yml
RUN pip install .
RUN django-admin.py collectstatic --noinput --settings "ganeti_webmgr.ganeti_web.settings"
RUN django-admin.py syncdb --noinput --settings "ganeti_webmgr.ganeti_web.settings"
```

To get the above dockerfile to work
1. go to the `python_webapp` directory
2. run `docker build -t osuosl/python_webapp .`
3. go to a clone of GWM and add the above to a Dockerfile.
4. Copy this into /path/to/gwm/config.yml: https://pastebin.osuosl.org/27691/ (It uses the secret keys I generated for the dockerfile before, so it's no big deal for them to be shared.)
5. `docker build -t osuosl/ganeti_webmgr .`
6. `docker run -p 80:80 osuosl/ganeti_webmgr`

And the docker file works its magic.

### CODE REVIEW

**Comment by pop** [comment]
Reviewed the code, tested it out, found some weird quirks:
- If the ports are set to 8000:80 it will redirect you to `http://localhost/accounts/login/?next=/`, which is wrong, if you manually change that to `http://localhost:8000/accounts/login/?next=/` the login screen shows up just fine.
- There are quite a bit of packages installed by default, I bet a few are gwm specific, we should try to keep this one as stripped down as possible.

Other than that it looks great :+1:

**Comment by tschuy** [comment]
Looking through the install list, most of those (except curl, which I'm removing with removing the ius repo) _are_ required:
1. python-devel, python-setuptools, gcc, libffi-devel are all required to do any pip package compilation. 
2. nginx is required (whaaat)
3. python-meld3 is required for supervisor

openssl-devel is the only optional one on that list. What do you think? I'll remove it for now.

**Comment by tschuy** [comment]
BTW, for anyone testing this, you'll want to use the dockerfile that exists on the GWM branch tschuy/dockerfile_update: https://github.com/osuosl/ganeti_webmgr/tree/tschuy/dockerfile_update

File: python_webapp/files/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+server {
+  listen 80 default_server;
+  access_log /var/log/nginx/access.log;
+  error_log /var/log/nginx/error.log;
+
+  location /static/ {
+    root /opt/app;
```
Comment (tschuy) [inline python_webapp/files/nginx.conf:7]: I'll add stuff to the README about these conveniences.

File: python_webapp/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+RUN pip install supervisor gunicorn
+RUN echo_supervisord_conf > /etc/supervisord.conf
+RUN printf "[include] \nfiles = /etc/supervisor.d/*.conf" >> /etc/supervisord.conf
+COPY files/nginx.conf /etc/nginx/conf.d/app.conf
+RUN rm -f /etc/nginx/conf.d/default.conf
+COPY files/gunicorn.conf /opt/app/conf/gunicorn_config.py
+COPY files/supervisor.conf /etc/supervisor.d/app.conf
+
+WORKDIR /opt/app/src
+CMD bash -c "nginx && supervisord && tail -f /var/log/nginx/access.log"
```
Comment (jordane) [inline python_webapp/Dockerfile:22]: Why not tail the error log too?

File: spec/python_webapp_spec.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+end
+
+%w(gunicorn supervisor).each do |pkg|
+  describe package(pkg) do
+    it { should be_installed.by('pip') }
+  end
+end
+
+describe command('cat /etc/redhat-release') do
+  its(:stdout) { should contain 'CentOS release 6' }
```
Comment (jordane) [inline spec/python_webapp_spec.rb]: What is the use-case for testing this?

Comment (iankronquist) [inline spec/python_webapp_spec.rb]: Test All the Things!
but seriously, this seems a little excessive.

Comment (tschuy) [inline spec/python_webapp_spec.rb]: I've removed it. I think it was mostly a first test I could have run successfully that didn't require the dockerfile doing anything complicated.

**Pull request closed without merging**